lib/optimize/InnerGraphPlugin.js

```diff
@@ -192,6 +205,20 @@ class InnerGraphPlugin {
 						if (parser.scope.topLevelScope === true) {
 							InnerGraph.setTopLevelSymbol(parser.state, undefined);
 
+							if (
+								pureExpressionStatements.has(
+									/** @type {StatementNode} */ (statement)
+								)
+							) {
+								const dep = new PureExpressionDependency(
+									/** @type {ExpressionStatementNode} */ (statement).expression.range
+								);
+								dep.loc = statement.loc;
+								dep.usedByExports = false;
+								parser.state.module.addDependency(dep);
+								return true;
+							}
```

Comment:
You need to `InnerGraph.setTopLevelSymbol` to something other than `undefined`, otherwise references in this statement won't be considered as unused.

`InnerGraph.setTopLevelSymbol(parser.state, false);` would make sense, but not sure if `false` is even handled there. This need to be added to:

* InnerGraph.onUsage need to differ between `false` and `undefined`
* InnerGraphPlugin parser.hooks.expression uses `InnerGraph.getTopLevelSymbol` and need to handle `false` too. (Not calling `InnerGraph.addUsage` at all)

Reply:
Not sure that I understand your correctly, because of `return true;` references in statement will not be visited. Anyway I refactored using `false` top level symbol